core/templates/dev/head/pages/exploration_player/ExplorationFooterDirective.js

```diff
@@ -44,6 +44,10 @@ oppia.directive('explorationFooter', [
             $scope.$apply();
           });
 
+          $scope.correctCount = function () {
```

Comment:
Functions should start with a verb.

Also, not clear what "correct count" is. Maybe rename as getNumberOf...()

Reply:
Okay! If a learner goes back to an earlier question, he/she isn't allowed to re-answer it.
So the count will remain the same as it was (including the point due to the one which we are revisiting).
![screenshot-2018-2-16 what is a fraction 1](https://user-images.githubusercontent.com/23234392/36300912-12d6988a-1329-11e8-8051-db2ee56f51d9.png)
 
I didn't try with numbers greater than 9, will add the screenshots for numbers greater than 9.

Reply:
I don't see a count in your screenshot.

Also, I wasn't talking about going back to an earlier question -- I was talking about looping back. I.e. the exploration itself contains a loop, and by progressing forward the learner ends up at a state that they were at before.

Reply:
Oh! Got it. The counter increments if there is a loop in the exploration(and we answer correctly). Shall it work the other way around?

Below are the screenshots(one with count less than 9 and one with greater than 9).

![screenshot-2018-2-16 personalized online learning from oppia 1](https://user-images.githubusercontent.com/23234392/36302231-b3ec4f86-132d-11e8-9454-ac44be8c43c5.png)

![screenshot-2018-2-16 personalized online learning from oppia](https://user-images.githubusercontent.com/23234392/36302232-b421f136-132d-11e8-8b73-976eef4abc4b.png)

Reply:
Thanks for the screenshots. The star is a bit lower down than it should be, I think, unless you've cut off part of the footer in the screenshot. (It looks cut off at the bottom edge.) Make sure to check on mobile, too. Also how does it interact with hints on desktop (which I believe appear in the bottom right)? You might want to ask Christine about this.

I think let's increment the counter for each newly-encountered question. It doesn't make sense for someone who loops around to get a higher counter than if they don't.

Reply:
Okay! Thanks!
I will adjust the alignment of the star as you said. The screenshots describe the situation in which the window is narrow(mobile). As you mentioned in the issue, this star is supposed to be there only in the case of mobile. Is there any specific way other than this(reducing the size of the window) to check for mobile? 

Regarding the hint part -- Hints are on the left bottom corner in case of mobile view(attached is the screenshot)

![screenshot-2018-2-16 personalized online learning from oppia 2](https://user-images.githubusercontent.com/23234392/36303967-fbfc43a2-1333-11e8-9551-196d99c6109f.png)

So for loops, although the question has been attempted it appears as if its a new question. How shall I keep track that the question has been answered once ?

Reply:
The Chrome browser has a way to emulate different mobile devices -- I suggest searching for details on that.

For the hints issue I was referring to the desktop view. Make sure to check both.

For loops -- I don't know, that's something you'll have to figure out as part of the impl. Maybe keep track of state names that have been answered correctly.

Reply:
Okay! I will look into that. 
There is a design in the Figma file for the desktop view, but in the issue ( #4364 ) its mentioned "Add a star showing counter (only on mobile)". So is there a need to check it for the desktop view?

Reply:
Good question. I don't know. You'll need to ask Christine what the intended design is, and perhaps flesh it out with her. A meeting might not be a bad idea.

Reply:
Okay! Will do the same.
Thanks!